libs/books/feature/src/lib/book-search/book-search.component.spec.ts

```diff
@@ -1,27 +1,75 @@
-import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import {
+  async,
+  ComponentFixture,
+  TestBed,
+  tick,
+  fakeAsync,
+  discardPeriodicTasks
+} from '@angular/core/testing';
 import { NoopAnimationsModule } from '@angular/platform-browser/animations';
-import { SharedTestingModule } from '@tmo/shared/testing';
-
+import { SharedTestingModule, createBook } from '@tmo/shared/testing';
 import { BooksFeatureModule } from '../books-feature.module';
 import { BookSearchComponent } from './book-search.component';
+import { provideMockStore, MockStore } from '@ngrx/store/testing';
+import { searchBooks } from '@tmo/books/data-access';
 
-describe('ProductsListComponent', () => {
+describe('BookSearchComponent', () => {
   let component: BookSearchComponent;
   let fixture: ComponentFixture<BookSearchComponent>;
+  let store: MockStore;
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
-      imports: [BooksFeatureModule, NoopAnimationsModule, SharedTestingModule]
+      imports: [
+        BooksFeatureModule,
+        NoopAnimationsModule,
+        SharedTestingModule
+      ],
+      providers: [
+        provideMockStore({
+          initialState: {
+            books: {
+              entities: []
+            }
+          }
+        }),
+      ],
     }).compileComponents();
   }));
 
   beforeEach(() => {
     fixture = TestBed.createComponent(BookSearchComponent);
     component = fixture.componentInstance;
     fixture.detectChanges();
+    store = TestBed.inject(MockStore);
   });
 
-  it('should create', () => {
-    expect(component).toBeDefined();
+  describe('ngOnInit()', () => {
+    beforeEach(() => {
+      spyOn(component.searchForm, 'valueChanges');
+      spyOn(store, 'dispatch');
+    });
+
+    it('searchBooks() should be called when we have empty value for search term and actual time spent after entering search term is equal to 500ms', fakeAsync(() => {
```

Comment:
this is confusing to read, also why should searchBooks be called when there is in an empty value for search term? (you're not even testing this ;)

Reply:
Corrected this.. Thanks